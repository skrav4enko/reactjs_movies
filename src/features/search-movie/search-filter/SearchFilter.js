import React from 'react';
import './SearchFilter.scss';
import InputField from '../../../shared/input-field/InputField';
import Button from '../../../shared/button/Button';
import ButtonGroup from '../../../shared/button-group/ButtonGroup';

function SearchFilter(props) {
  return (
    <div className="search-filter">
      <div className="container">
        <h1 className="search-filter__title">Find your movie</h1>
        <div className="search-filter__search">
          <InputField placeholder="Search" /> <Button>Search</Button>
        </div>
        <div>
          <span className="search-filter__search-by">Search by</span>
          <ButtonGroup>
            <Button size="small">Title</Button>
            <Button variant="secondary" size="small">
              Genre
            </Button>
          </ButtonGroup>
        </div>
      </div>
    </div>
  );
}

export default SearchFilter;
