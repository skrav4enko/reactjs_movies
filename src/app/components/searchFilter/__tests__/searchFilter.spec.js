/* eslint-disable */
import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchFilter from '..';

describe('SearchFilter', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      filterBy: 'title',
      handleFilter: jest.fn(),
      handleFilterBy: jest.fn(),
    };
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<SearchFilter {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should has input value', () => {
    const newValue = 'new Value';
    const testProps = {
      ...defaultProps,
      filterBy: 'genres',
    };

    const component = mount(<SearchFilter {...testProps} />);
    const input = component.find('input');

    input.simulate('change', {target: { value: newValue}});
    expect(component.state().value).toBe(newValue);
  });

  it('should call submit', () => {
    const testProps = {
      ...defaultProps,
      filterBy: 'genres',
    };

    const wrapper = mount(<SearchFilter {...testProps} />);
    const form = wrapper.find('form');
    const instance = wrapper.instance();

    jest.spyOn(instance, 'resetState');
    form.simulate('submit');

    expect(instance.resetState).toHaveBeenCalled();
  });

  it('should match snapshot 2', () => {
    const testProps = {
      ...defaultProps,
      filterBy: 'genres',
    };

    const wrapper = shallow(<SearchFilter {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleFilterBy with genres', () => {
    const component = mount(<SearchFilter {...defaultProps} />);
    const button = component.find('.button.small.secondary').simulate('click');
    expect(defaultProps.handleFilterBy).toHaveBeenCalledWith('genres');
  });

  it('should call handleFilterBy with title', () => {
    const testProps = {
      ...defaultProps,
      filterBy: 'genres',
    };

    const component = mount(<SearchFilter {...testProps} />);
    const button = component.find('.button.small.secondary').simulate('click');
    expect(testProps.handleFilterBy).toHaveBeenCalledWith('title');
  });
});
