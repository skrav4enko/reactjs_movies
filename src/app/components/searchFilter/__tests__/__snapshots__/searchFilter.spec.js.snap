// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter should match snapshot 1`] = `
<index.tsx
  filterBy="title"
  handleFilter={[MockFunction]}
  handleFilterBy={[MockFunction]}
/>
`;

exports[`SearchFilter should match snapshot 2 1`] = `
<index.tsx
  filterBy="genres"
  handleFilter={[MockFunction]}
  handleFilterBy={[MockFunction]}
/>
`;
